### ISSUE TITLE
Memory Leak on Achievements implementations. 

### ISSUE THREAD

**Issue by tasomaniac** [gdg-x/frisbee#447]
Achievement implementation immediately leaks Activity on open. 
It happens both on SettingsActivity and AboutActivity. 

I think it is because of the delay with Handler implementation. It makes also the app crash a lot. I think it is the most common unresolved crash of the app in Crashlytics as of now. 

```
In org.gdg.frisbee.android.debug:2.2.1-debug:22100.
* org.gdg.frisbee.android.activity.SettingsActivity has leaked:
* GC ROOT com.google.android.gms.games.internal.GamesClientImpl$PopupLocationInfoBinderCallbacks.zzaoU
* references com.google.android.gms.games.internal.PopupManager$PopupManagerHCMR1.zzaqt
* references com.google.android.gms.games.internal.GamesClientImpl.mContext
* leaks org.gdg.frisbee.android.activity.SettingsActivity instance

* Reference Key: 06c83aca-47c0-481a-8d31-e24a87b6c913
* Device: motorola google Nexus 6 shamu
* Android Version: 5.1.1 API: 22 LeakCanary: 1.3.1
* Durations: watch=5023ms, gc=174ms, heap dump=16281ms, analysis=27519ms

* Details:
* Instance of com.google.android.gms.games.internal.GamesClientImpl$PopupLocationInfoBinderCallbacks
|   zzaoU = com.google.android.gms.games.internal.PopupManager$PopupManagerHCMR1 [id=0x33643fc0]
|   mDescriptor = java.lang.String [id=0x32c58240]
|   mObject = -2008139912
|   mOwner = com.google.android.gms.games.internal.GamesClientImpl$PopupLocationInfoBinderCallbacks [id=0x336af480]
* Instance of com.google.android.gms.games.internal.PopupManager$PopupManagerHCMR1
|   zzaoV = false
|   zzaqx = java.lang.ref.WeakReference [id=0x33644040]
|   zzaqt = com.google.android.gms.games.internal.GamesClientImpl [id=0x335e9780]
|   zzaqu = com.google.android.gms.games.internal.PopupManager$PopupLocationInfo [id=0x33642700]
* Instance of com.google.android.gms.games.internal.GamesClientImpl
|   zzaoQ = com.google.android.gms.games.internal.GamesClientImpl$1 [id=0x336607d0]
|   zzaoR = java.lang.String [id=0x32e86c80]
|   zzaoS = com.google.android.gms.games.PlayerEntity [id=0x335a1920]
|   zzaoT = com.google.android.gms.games.GameEntity [id=0x33565180]
|   zzaoU = com.google.android.gms.games.internal.PopupManager$PopupManagerHCMR1 [id=0x33643fc0]
|   zzaoV = false
|   zzaoW = android.os.Binder [id=0x33643fa0]
|   zzaoX = 512942283
|   zzaoY = com.google.android.gms.games.Games$GamesOptions [id=0x336426a0]
|   mContext = org.gdg.frisbee.android.activity.SettingsActivity [id=0x32ddc4c0]
|   mHandler = com.google.android.gms.common.internal.zzi$zzb [id=0x33643f40]
|   zzMY = null
|   zzWJ = java.util.Collections$UnmodifiableSet [id=0x336605f0]
|   zzWt = android.os.Looper [id=0x32c02040]
|   zzXa = com.google.android.gms.common.internal.zze [id=0x336424c0]
|   zzaak = com.google.android.gms.common.internal.zzl [id=0x32fa4ba0]
|   zzaal = com.google.android.gms.common.internal.zzp$zza$zza [id=0x3366a430]
|   zzaam = com.google.android.gms.common.api.zze$zzc [id=0x3366ca80]
|   zzaan = null
|   zzaao = java.util.ArrayList [id=0x33643f20]
|   zzaap = null
|   zzaaq = 1
|   zzaar = com.google.android.gms.common.api.zzg$2 [id=0x33660680]
|   zzaas = com.google.android.gms.common.api.zzg$4 [id=0x33643ee0]
|   zzaat = 1
|   zzaau = java.util.concurrent.atomic.AtomicInteger [id=0x33660770]
|   zzqt = java.lang.Object [id=0x33660760]
* Instance of org.gdg.frisbee.android.activity.SettingsActivity
|   mAchievementActionHandler = org.gdg.frisbee.android.achievements.AchievementActionHandler [id=0x33644280]
|   mActionBarToolbar = android.support.v7.widget.Toolbar [id=0x33491400]
|   mGoogleApiClient = com.google.android.gms.common.api.zzg [id=0x335e9700]
|   mHandler = android.os.Handler [id=0x335fc900]
|   mSignInIntent = null
|   mSignInProgress = 0
|   scopedBus = org.gdg.frisbee.android.utils.ScopedBus [id=0x336600d0]
|   mCurrentPage = 0
|   mDelegate = android.support.v7.app.AppCompatDelegateImplV14 [id=0x33521a60]
|   mAllLoaderManagers = android.support.v4.util.SimpleArrayMap [id=0x3366c3e0]
|   mCheckedForLoaderManager = true
|   mContainer = android.support.v4.app.FragmentActivity$2 [id=0x336600c0]
|   mCreated = true
|   mFragments = android.support.v4.app.FragmentManagerImpl [id=0x3361c0b0]
|   mHandler = android.support.v4.app.FragmentActivity$1 [id=0x335fc8e0]
|   mLoaderManager = null
|   mLoadersStarted = false
|   mOptionsMenuInvalidated = false
|   mReallyStopped = true
|   mResumed = false
|   mRetaining = false
|   mStopped = true
|   mActionBar = null
|   mActivityInfo = android.content.pm.ActivityInfo [id=0x335c9080]
|   mActivityTransitionState = android.app.ActivityTransitionState [id=0x335fb480]
|   mAllLoaderManagers = android.util.ArrayMap [id=0x3366c3c0]
|   mApplication = org.gdg.frisbee.android.app.App [id=0x32e7d3a0]
|   mCalled = true
|   mChangeCanvasToTranslucent = false
|   mChangingConfigurations = false
|   mCheckedForLoaderManager = true
|   mComponent = android.content.ComponentName [id=0x3358c100]
|   mConfigChangeFlags = 0
|   mContainer = android.app.Activity$1 [id=0x33660090]
|   mCurrentConfig = android.content.res.Configuration [id=0x335005e0]
|   mDecor = null
|   mDefaultKeyMode = 0
|   mDefaultKeySsb = null
|   mDestroyed = true
|   mDoReportFullyDrawn = false
|   mEmbeddedID = null
|   mEnableDefaultActionBarUp = false
|   mEnterTransitionListener = android.app.SharedElementCallback$1 [id=0x70fbc220]
|   mExitTransitionListener = android.app.SharedElementCallback$1 [id=0x70fbc220]
|   mFinished = true
|   mFragments = android.app.FragmentManagerImpl [id=0x334e8f90]
|   mHandler = android.os.Handler [id=0x335fc8c0]
|   mIdent = 880166363
|   mInstanceTracker = android.os.StrictMode$InstanceTracker [id=0x336600a0]
|   mInstrumentation = android.app.Instrumentation [id=0x32e99600]
|   mIntent = android.content.Intent [id=0x334af080]
|   mLastNonConfigurationInstances = null
|   mLoaderManager = null
|   mLoadersStarted = false
|   mMainThread = android.app.ActivityThread [id=0x32c03100]
|   mManagedCursors = java.util.ArrayList [id=0x335fc880]
|   mManagedDialogs = null
|   mMenuInflater = null
|   mParent = null
|   mReferrer = java.lang.String [id=0x3362f900]
|   mResultCode = 0
|   mResultData = null
|   mResumed = false
|   mSearchManager = null
|   mStartedActivity = false
|   mStopped = true
|   mTemporaryPause = false
|   mTitle = java.lang.String [id=0x32dd2180]
|   mTitleColor = 0
|   mTitleReady = true
|   mToken = android.os.BinderProxy [id=0x3362f520]
|   mTranslucentCallback = null
|   mUiThread = java.lang.Thread [id=0x741a4000]
|   mVisibleBehind = false
|   mVisibleFromClient = true
|   mVisibleFromServer = true
|   mVoiceInteractor = null
|   mWindow = com.android.internal.policy.impl.PhoneWindow [id=0x32eafb00]
|   mWindowAdded = true
|   mWindowManager = android.view.WindowManagerImpl [id=0x335fcb40]
|   mInflater = com.android.internal.policy.impl.PhoneLayoutInflater [id=0x335f9280]
|   mOverrideConfiguration = null
|   mResources = android.content.res.Resources [id=0x32dbd820]
|   mTheme = android.content.res.Resources$Theme [id=0x335fcb60]
|   mThemeResource = 2131362034
|   mBase = android.app.ContextImpl [id=0x335e9500]
```

**Comment by altamish1994** [comment]
I want to contribute to the app and fix this issue but when I execute the code and run the app on my device it  says no API key defined and when i signup through settings the app crashes continuously, can you guide me how to get through this

**Comment by tasomaniac** [comment]
We have to fix that crash. Even when you don't have the keys, it shouldn't
crash.

Can you reach us through gitter.im. You can find the link on the Readme
file. I can help you there.

**Comment by friedger** [comment]
using applicationContext as in https://github.com/gdg-x/frisbee/pull/560/files#diff-cb1c64b03b5c616289ab49c37855066fR136 fixes it. Is this the best way to do it?

**Comment by friedger** [comment]
fixed by using application context (see #560 )
